jwt/validate.go

```diff
@@ -352,17 +362,41 @@ func (ccs claimContainsString) Validate(_ context.Context, t Token) ValidationEr
 		return NewValidationError(fmt.Errorf(`claim %q must be a []string (got %T)`, ccs.name, v))
 	}
 
-	var found bool
 	for _, v := range list {
 		if v == ccs.value {
-			found = true
-			break
+			return nil
 		}
 	}
-	if !found {
-		return NewValidationError(fmt.Errorf(`%q not satisfied`, ccs.name))
+	return NewValidationError(fmt.Errorf(`%q not satisfied`, ccs.name))
+}
+
+type audienceClaimContainsString struct {
+	value string
+}
+
+// AudienceClaimContainsString can be used to check if the audience claim, which is
+// expected to be a list of strings, contains `value`.
+func AudienceClaimContainsString(value string) Validator {
+	return audienceClaimContainsString{value: value}
+}
+
+func (ccs audienceClaimContainsString) Validate(_ context.Context, t Token) ValidationError {
+	v, ok := t.Get(AudienceKey)
+	if !ok {
+		return NewValidationError(fmt.Errorf(`claim %q not found`, AudienceKey))
```

Comment:
I don't understand why this is not of type `errInvalidAudience`. If the claim doesn't exist, then the audience value is invalid, right?

Reply:
Yes, this actually answer the question I asked in the PR's description. I agree that it should return `errInvalidAudience`, but wanted to be sure before making more changes